src/method.jl

```diff
@@ -91,14 +91,22 @@ function PercivalSolver(nlp::AbstractNLPModel{T, V}) where {T, V}
 end
 
 @doc (@doc PercivalSolver) function percival(::Val{:equ}, nlp::AbstractNLPModel; kwargs...)
+  if !(nlp.meta.minimize)
+    error("Percival only works for minimization problem")
+  end
+  if nlp.meta.ncon == 0 || !equality_constrained(nlp)
+    error(
+      "percival(::Val{:equ}, nlp) should only be called for equality-constrained problems with bounded variables. Use percival(nlp)",
+    )
+  end
   solver = PercivalSolver(nlp)
-  solve!(Val(:equ), solver, nlp; kwargs...)
+  SolverCore.solve!(solver, nlp; kwargs...)
 end
 
-function solve!(
-  ::Val{:equ},
+function SolverCore.solve!(
```

Comment:
I think the other `percival` call in lines 7, 17, and 46 also need to be changed. I think `solve` was introduced without fixing this in the past. I think the proper way is

`solve!(solver, ...)` in line 7 decides which specific `solve!(Val{...}, ...)` to call.
`percival` on line 93 creates `solver` and calls `solve!(solver, ...)`.

Then, possibly lines 94 and 101 are not needed.

Reply:
I am not sure this would work. The issue is that the `PercivalSolver` is only valid for the `equ` case. For `ineq` you need an extended version of `PercivalSolver` and for `tron` it is a `TronSolver`.

Reply:
You're right, let's leave it like this then.